concept/thing/thing_manager/validation/mod.rs

```diff
@@ -0,0 +1,160 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use std::{error::Error, fmt};
+
+use encoding::value::{label::Label, value::Value, value_type::ValueType};
+
+use crate::{
+    error::ConceptReadError,
+    thing::{attribute::Attribute, object::Object, relation::Relation},
+    type_::{
+        annotation::{AnnotationCardinality, AnnotationRange, AnnotationRegex, AnnotationValues},
+        attribute_type::AttributeType,
+        object_type::ObjectType,
+        owns::Owns,
+        plays::Plays,
+        relates::Relates,
+        role_type::RoleType,
+        Ordering,
+    },
+};
+
+pub(crate) mod commit_time_validation;
+pub(crate) mod operation_time_validation;
+pub(crate) mod validation;
+
+#[derive(Debug, Clone)]
+pub enum DataValidationError {
```

Comment:
Moved a bunch of ConceptWriteErrors related to validation here and added much more new errors.